service/api/api.go

```diff
@@ -0,0 +1,70 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/rodrigomkd/go-rest-api/model"
+)
+
+type IApiService interface {
+	GetActivities() []model.Activity
+}
+
+type ApiService struct {
+	activitiesUri string
+	api           IApiService
+}
+
+func New(activitiesUri string) *ApiService {
+	return &ApiService{
+		activitiesUri: activitiesUri,
+	}
+}
+
+//ReadAPI - call api and returns the response
+func readAPI(uri string) []byte {
+	response, err := http.Get(uri)
```

Comment:
I would recommend taking a look at this https://www.thegreatcodeadventure.com/mocking-http-requests-in-golang/

When testing this, we want to be able to mock the http call instead of actually calling an external server, just like how we want to read a different csv file when testing that service.

I would suggest using the `*http.Client` structure as an attribute of your service struct first. Then you could use that client's Get method instead. Once that is done you could define an interface that holds the clients functions that you're using, then you could replace the httpClient attribute with that interface and receive it as a param on your New function. Then on the main func, when creating this service you could receive that &http.Client{} value.

This way you can mock the http client and avoid doing actual http calls when doing unit testing